frontend/src/routes/manage/shared.tsx

```diff
@@ -0,0 +1,575 @@
+            tableLayout: "fixed",
+            "& > thead": {
+                position: "sticky",
+                top: 0,
+                zIndex: 10,
+                backgroundColor: COLORS.neutral05,
+                "&  > tr > th": {
+                    borderBottom: `1px solid ${COLORS.neutral25}`,
+                    textAlign: "left",
+                    padding: "8px 12px",
+                },
+                ...headerSticks && {
+                    boxShadow: "0 0 20px rgba(0, 0, 0, 0.3)",
+                    clipPath: "inset(0px 0px -20px 0px)",
+                },
+            },
+            "& > tbody": {
+                "& > tr:hover, tr:focus-within": {
+                    backgroundColor: COLORS.neutral15,
+                },
+                "& > tr:not(:first-child) > td": {
+                    borderTop: `1px solid ${COLORS.neutral25}`,
+                },
+                "& td": {
+                    padding: 6,
+                    verticalAlign: "top",
+                    "&:not(:first-child)": {
+                        padding: "8px 12px 8px 8px",
+                    },
+                },
+            },
+        }}>
+            <colgroup>
+                {/* Each table has thumbnails, but their width might vary */}
+                <col span={1} css={{ width: (thumbnailWidth ?? THUMBNAIL_WIDTH) + 2 * 6 }} />
+                {/* Each table has a column for title and description */}
+                <col span={1} />
+                {/*
+                    Additional columns can be declared in the specific column array.
+                */}
+                {additionalColumns?.map(col =>
+                    <col key={col.key} span={1} css={{ width: col.headerWidth ?? 135 }} />)
+                }
+            </colgroup>
+
+            <thead ref={tableHeaderRef}>
+                <tr>
+                    {/* Thumbnail */}
+                    <th></th>
+                    {/* Title */}
+                    <ColumnHeader
+                        label={t("manage.item-table.columns.title")}
+                        sortKey="TITLE"
+                        {...{ vars }}
+                    />
+                    {/* Sort columns */}
+                    {additionalColumns?.map(col => (
+                        <ColumnHeader
+                            key={col.key}
+                            label={t(col.label)}
+                            sortKey={col.key}
+                            {...{ vars }}
+                        />
+                    ))}
+                </tr>
+            </thead>
+            <tbody>
+                {connection.__typename === "EventConnection" && connection.items.map(event =>
+                    <EventRow key={event.id} event={event} />)
+                }
+                {connection.__typename === "SeriesConnection" && connection.items.map(series =>
+                    <SeriesRow key={series.id} series={series} />)
+                }
```

Comment:
Mhh I don't think that's too pretty. For one, it reads weird query response things, but more importantly, the data flow between components and files is a bit weird. Both "callers" of `<ManageItems>` pass all data to that component, but then randomly, the component deep inside imports another component from the "calling" file. I think it would be better if the series/event route would pass this component to `ManageItems`. So add a `RenderRow: (item: T) => ReactNode;` prop to `ManageItem`. Then it needs to be generic, yes, and also some of the helper sub-components. But I think this will also actually add nice typing to lots of components. No more `type SortColumn = VideosSortColumn | SeriesSortColumn;` and the like. At this place you would just do `<RenderRow ... />`.